File: Exemplos/App_IntentDigitalHUB_XamarinForms/Elgin_IntentDigitalHUB_XamarinForms-font/E1_Digital_Hub_Xamarin_Forms_M8/IntentServices/Bridge/IniciaVendaDebito.cs
namespace Xamarin_Forms_Intent_Digital_Hub.IntentServices.Bridge
{
    class IniciaVendaDebito : BridgeCommand
    {
        readonly int idTransacao;
        readonly string pdv;
        readonly string valorTotal;

        public IniciaVendaDebito(int idTransacao, string pdv, string valorTotal) : base("IniciaVendaDebito")
        {
            this.idTransacao = idTransacao;
            this.pdv = pdv;
            this.valorTotal = valorTotal;
        }

        protected override string FunctionParameters()
        {
            return "\"idTransacao\"" + ":" + idTransacao + "," +
                    "\"pdv\"" + ":" + "\"" + pdv + "\"" + "," +
                    "\"valorTotal\"" + ":" + "\"" + valorTotal + "\"";
        }
    }
}

File: Exemplos/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/BRIDGE/IniciaVendaDebito.ts
import {BridgeCommand} from './BridgeCommand';

export class IniciaVendaDebito extends BridgeCommand {
  readonly idTransacao: number;
  readonly pdv: string;
  readonly valorTotal: string;

  constructor(idTransacao: number, pdv: string, valorTotal: string) {
    super('IniciaVendaDebito');
    this.idTransacao = idTransacao;
    this.pdv = pdv;
    this.valorTotal = valorTotal;
  }

  functionParameters(): object {
    return {
      idTransacao: this.idTransacao,
      pdv: this.pdv,
      valorTotal: this.valorTotal,
    };
  }
}
